### PR TITLE
Implement myAtoi in Rust

### PR DESCRIPTION
**Pull request by MohsenBg**
### Description
This PR adds a Rust implementation of the myAtoi function (similar to C's atoi), which converts a string to a 32-bit signed integer following specific parsing rules. It includes robust error handling and adheres to the constraints of the LeetCode problem "8. String to Integer (atoi)".

### Features
- Handles leading whitespace and optional sign
- Parses digits and stops at first non-digit
- Prevents overflow using `checked_mul` and `checked_add`
- Includes unit tests for edge cases and common inputs

### CODE REVIEW

File: MohsenBg/string_to_integer_atoi/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        } else {
+            // Non-digit character found after number started: stop parsing
+            break;
+        }
+    }
+
+    num * sign // Apply sign and return final result
+}
+
+#[cfg(test)]
```
Comment (Rezishon) [inline MohsenBg/string_to_integer_atoi/src/lib.rs:42]: Beautiful man. Well done <3

**Review by Rezishon — COMMENTED** [review]
Very clear and nice code. So so well done buddy.
